api/envoy/config/route/v3/route_components.proto

```diff
@@ -760,6 +760,29 @@ message RouteAction {
     ConnectConfig connect_config = 3;
   }
 
+  message MaxStreamDuration {
+    // Specifies the maximum duration allowed for streams on the route. If not specified, the
+    // value from :ref:`max_stream_duration
+    // <envoy_api_field_config.core.v3.HttpProtocolOptions.max_stream_duration>` is used.
+    google.protobuf.Duration max_stream_duration = 1;
+
+    // If present, and the request is a gRPC request, use the
```

Comment:
We should document how it is determined that the request is a gRPC request.

Reply:
Done.